src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -682,27 +682,19 @@ public void setQuorumAddress(InetSocketAddress addr){
     }
 
     public InetSocketAddress getElectionAddress(){
-        synchronized (QV_LOCK) {
-            return myElectionAddr;
-        }
+        return myElectionAddr;
```

Comment:
Note this synchronization was introduced as part of ZOOKEEPER-2080 (which fixed a separate dead lock issue.). The intent was to protect access on these states while a reconfig operation is in flight, but my analysis indicate it might be totally fine without these synchronizations.